### PR TITLE
[WIP] Selectors separation

### PR DESCRIPTION
**Pull request by umrashrf**
Please do not merge this. I am yet to do following things.

For the sake of this PR, please find temporary [`selectors`](http://github.com/umrashrf/selectors) repo.

- I am working on separating tests. There are selector tests which has to be kept back because of `HtmlResponse`, `XmlResponse`, `TextResponse` classes. And there are tests which can be migrated to `selectors`.
- Probably adding deprecation notices for `scrapy.selector.Selector`, to use `selectors.Selector` instead. Not sure if this should be really done. Please enlighten me.
- Remove dependencies used solely by selectors

Above all I need your valuable comments.

### CODE REVIEW

**Comment by umrashrf** [comment]
Tests are separated.

Changes for #906

**Comment by nyov** [comment]
I think this would be a perfect time and place to remove all the `@deprecated` code in here at once?
Or if the scrapy repository wishes to keep them for some time still, they should be re-added or kept there (as wrappers).

**Comment by kmike** [comment]
I think it looks good.

@umrashrf - what is missing?

**Comment by umrashrf** [comment]
@nyov @kmike sorry for late reply, this is weird I didn't get notified.

@nyov you're right. I was going to drop it.

@kmike Script delivery is pending https://github.com/scrapy/scrapy/issues/906#issuecomment-71879184.

**Comment by umrashrf** [comment]
@nyov @kmike updated selectors repo and this PR based on changes in master.

Should I drop `scrapy/docs/topics/selectors.rst` in favor of `selectors/docs/topics/selectors.rst`?

**Comment by kmike** [comment]
I think we shouldn't drop selector docs entirely. We can adjust docs and maybe remove some reference parts, but I think having a selectors "tutorial" in Scrapy docs is a good idea.

File: requirements.txt [hunk 1]
```diff
@@ -5,3 +5,4 @@ cssselect>=0.9
 w3lib>=1.8.0
 queuelib
 six>=1.5.2
+git+https://github.com/umrashrf/selectors.git#egg=selectors
```
Comment (nramirezuy) [inline requirements.txt]: Is is going to live on umair repo?

**Comment by kmike** [comment]
As @pablohoffman noted, selectors is not a good name because of https://docs.python.org/3/library/selectors.html

**Comment by Curita** [comment]
Some decisions made by maintainers so we can move forward with separating scrapy selectors:
* We can't use either `selectors` or `requests` as the new package name since those are already taken. I created a repo here https://github.com/scrapy/scrapy-selectors so we can move @umrashrf's code there (we can change the name later).
* We won't remove the current docs about selectors from Scrapy right now. Instead, we're going to keep a simple README file in the new package linking to the docs in Scrapy, with some simple snippets showing the selectors' usage.
* About git history: we're going to use the one created with `git filter-branch`.

**Comment by dangra** [comment]
Please see https://github.com/scrapy/scrapy-selectors/issues/1

**Comment by umrashrf** [comment]
thanks @Curita for summing it up. What about https://github.com/scrapy/scrapy/issues/586#issuecomment-94919509? I think it's a good point because without selector, scrapy would just be a crawling tool.

**Comment by nyov** [comment]
> What about https://github.com/scrapy/scrapy/issues/586#issuecomment-94919509?

What about it? What's the context for mentioning it?
Actually I don't agree with the comment, scrapy is still a data extraction tool without Selectors, to me. They are non-essential and people also use regex', beautiful soup and other libraries on source content.  Extraction logic may also be elsewhere, say the Image pipeline, so I disagree.

> As @pablohoffman noted, selectors is not a good name [...]

An alternative name - that is hard. Maybe defining the project scope first is important for that. Would we consider expanding this to binary format "selectors" for example? Is it considered to be a parser library, a data extraction kit or a bunch of convenience wrappers? Just some thoughts.

**Comment by Digenis** [comment]
Isn't this ticket meant to be closed to be continued in #1409 ?

**Comment by kmike** [comment]
@Digenis you're right, thanks!

**Pull request closed without merging**